discovery-provider/src/queries/get_save_tracks.py

```diff
@@ -41,100 +45,205 @@ class GetSaveTracksArgs(TypedDict):
+                        "Dy Mon DD YYYY HH24:MI:SS",
+                    ),
+                    TracksTable.created_at,
                 )
             )
-
-        if sort_method == SortMethod.title:
-            base_query = base_query.order_by(sort_fn(TrackWithAggregates.title))
-        elif sort_method == SortMethod.artist_name:
-            base_query = base_query.join(
-                TrackWithAggregates.user, aliased=True
-            ).order_by(sort_fn(User.name))
-        elif sort_method == SortMethod.release_date:
+        )
+    elif sort_method == SortMethod.plays:
+        base_query = base_query.join(TracksTable.aggregate_play).order_by(
+            sort_fn(AggregatePlay.count)
+        )
+    elif sort_method == SortMethod.reposts:
+        base_query = base_query.join(TracksTable.aggregate_track).order_by(
+            sort_fn(AggregateTrack.repost_count)
+        )
+    elif sort_method == SortMethod.saves:
+        base_query = base_query.join(TracksTable.aggregate_track).order_by(
+            sort_fn(AggregateTrack.save_count)
+        )
+    else:
+        # This branch covers added_date, or any other sort method
+        if filter_type == LibraryFilterType.favorite:
             base_query = base_query.order_by(
-                sort_fn(
-                    coalesce(
-                        func.to_date_safe(
-                            TrackWithAggregates.release_date,
-                            "Dy Mon DD YYYY HH24:MI:SS",
-                        ),
-                        TrackWithAggregates.created_at,
-                    )
-                )
+                sort_fn(Save.created_at), desc(TracksTable.track_id)
             )
-        elif sort_method == SortMethod.added_date:
+        elif filter_type == LibraryFilterType.repost:
             base_query = base_query.order_by(
-                sort_fn(Save.created_at), desc(TrackWithAggregates.track_id)
-            )
-        elif sort_method == SortMethod.plays:
-            base_query = base_query.join(TrackWithAggregates.aggregate_play).order_by(
-                sort_fn(AggregatePlay.count)
-            )
-        elif sort_method == SortMethod.reposts:
-            base_query = base_query.join(TrackWithAggregates.aggregate_track).order_by(
-                sort_fn(AggregateTrack.repost_count)
+                sort_fn(Repost.created_at), desc(TracksTable.track_id)
             )
-        elif sort_method == SortMethod.saves:
-            base_query = base_query.join(TrackWithAggregates.aggregate_track).order_by(
-                sort_fn(AggregateTrack.save_count)
+        elif filter_type == LibraryFilterType.purchase:
+            base_query = base_query.order_by(
+                sort_fn(USDCPurchase.created_at), desc(TracksTable.track_id)
             )
-        else:
+        elif filter_type == LibraryFilterType.all:
             base_query = base_query.order_by(
-                desc(Save.created_at), desc(TrackWithAggregates.track_id)
```

Comment:
This was the old implicit sort behavior; if no sort method specified, always sort desc. Made this more explicit in this refactor